src/storage/ducklake_transaction.cpp

```diff
@@ -476,9 +477,24 @@ void DuckLakeTransaction::CheckForConflicts(const TransactionChangeInformation &
 	for (auto &table_id : changes.tables_deleted_from) {
 		ConflictCheck(table_id, other_changes.dropped_tables, "delete from table", "dropped it");
 		ConflictCheck(table_id, other_changes.altered_tables, "delete from table", "altered it");
-		ConflictCheck(table_id, other_changes.tables_deleted_from, "delete from table", "deleted from it");
 		ConflictCheck(table_id, other_changes.tables_compacted, "delete from table", "compacted it");
 	}
+	if (!changes.tables_deleted_from.empty()) {
+		// If we have deletes on the tables check for files being deleted
+		const auto deleted_files = metadata_manager->GetFilesDeletedOrDroppedAfterSnapshot(transaction_snapshot);
```

Comment:
Before we run this query can we first check if we have any potential conflicts based on the `tables_deleted_from`? If there is no overlap we can skip this query.